terraform/nr-alerts/nr-alerts.tf

```diff
@@ -1,77 +1,14 @@
-#############################################
-# EC2 Instances
-#############################################
-
-### Fetch Canary EC2 Instances IDs by TAG.
-data "aws_instances" "canary_instances" {
-  filter {
-    name   = "tag:Name"
-    values = [var.instance_name_pattern]
-  }
-}
-
-### Fetch EC2 Instances by IDs.
-data "aws_instance" "canary_instance" {
-  count       = length(data.aws_instances.canary_instances.ids)
-  instance_id = data.aws_instances.canary_instances.ids[count.index]
-}
-
-#############################################
-# Alerts
-#############################################
-
-# Iterate all the instances and prepare alerts name <-> instance name relationship.
-# i.e:
-# alerts_instances = {
-# "canary:v1.37.2:amd64:al-2"            = "policies_prefix / al-2 / amd64"
-# "canary:v1.37.2:amd64:al-2022"         = "policies_prefix / al-2022 / amd64"
-# "canary:v1.37.2:amd64:centos-stream"   = "policies_prefix / centos-stream / amd64"
-# "canary:v1.37.2:amd64:centos7"         = "policies_prefix / centos7 / amd64"
-# }
-locals {
-  alerts_instances = {
-    for ins in data.aws_instance.canary_instance :
-    ins.tags.Name => can(regex(".+:.+:.+:.+", ins.tags.Name)) ?
-    (join(" / ", [
-      var.policies_prefix,
-      split(":", ins.tags.Name)[3],
-      split(":", ins.tags.Name)[2]
-    ])) :
-    (ins.tags.Name)
-  }
-}
-
-# Uncomment this to "debug" the generated structure
-output prueba {
-  value = local.alerts_instances
-}
-
-# Iterate alerts_instances and create policies for Canary EC2 instance.
-# i.e:
-# alerts_instances = {
-# "canary:v1.37.2:amd64:al-2"            = "policies_prefix / al-2 / amd64"
-# "canary:v1.37.2:amd64:al-2022"         = "policies_prefix / al-2022 / amd64"
-# "canary:v1.37.2:amd64:centos-stream"   = "policies_prefix / centos-stream / amd64"
-# "canary:v1.37.2:amd64:centos7"         = "policies_prefix / centos7 / amd64"
-# }
-# Will create policies:
-# [
-# "policies_prefix / al-2 / amd64"
-# "policies_prefix / al-2022 / amd64"
-# "policies_prefix / centos-stream / amd64"
-# "policies_prefix / centos7 / amd64"
-# ]
-resource "newrelic_alert_policy" "alert_canary" {
-  for_each = local.alerts_instances
-  name     = each.value
+resource "newrelic_alert_policy" "alert_canary_v2" {
```

Comment:
Why is the name v2 if you removed the old alert_canary resource? Does it need the old one to keep existing for som terraform state or something?